sdk/jsonexpr/operators/match_operator.py

```
@@ -1,15 +1,55 @@
+                    logger.warning(f"Regex pattern too long ({len(pattern)} chars), rejecting")
+                    return None
+
+                try:
+                    compiled = re.compile(pattern)
+                    with timeout(self.REGEX_TIMEOUT_SECONDS):
+                        return bool(compiled.match(text))
+                except re.error as e:
+                    logger.warning(f"Invalid regex pattern: {e}")
+                    return None
+                except TimeoutError:
+                    logger.warning(f"Regex execution timeout (potential ReDoS)")
```

Ruff (0.15.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)